server/config/service.go

```diff
@@ -85,18 +90,25 @@ func (s *service) newConfig() Config {
 	}
 }
 
-func (s *service) newInitializedConfig(newStoredMap map[string]interface{}, log utils.Logger) (*Config, error) {
+func (s *service) newInitializedConfig(newStoredConfig StoredConfig) (*Config, utils.Logger, error) {
+	log := s.NewBaseLogger()
 	conf := s.newConfig()
+	conf.StoredConfig = newStoredConfig
 	newMattermostConfig := s.reloadMattermostConfig()
-	license := s.getMattermostLicense(log)
+
+	if conf.DeveloperModeOverride != nil {
+		conf.DeveloperMode = *conf.DeveloperModeOverride
+	} else {
+		conf.DeveloperMode = pluginapi.IsConfiguredForDevelopment(newMattermostConfig)
+	}
```

Comment:
Is it possible to revert the `DeveloperModeOverride`, i.e. the value made `nil` again? `override === false` shouldn't necessarily mean that we don't want developer mode on right?

Reply:
Yes, and because DeveloperModeOverride is a stored config value, a change to it is propagated in OnConfigurationChanged, and `DeveloperMode` is set to the "server" value.